src/Database/Type/DateType.php

```diff
@@ -157,22 +151,19 @@ public function manyToPHP(array $values, array $fields, Driver $driver): array
      * Convert request data into a datetime object.
      *
      * @param mixed $value Request data
-     * @return \Cake\Chronos\ChronosDate|\DateTimeInterface|null
+     * @return \Cake\Chronos\ChronosDate|null
      */
-    public function marshal(mixed $value): ChronosDate|DateTimeInterface|null
+    public function marshal(mixed $value): ?ChronosDate
     {
-        if ($value instanceof DateTimeInterface || $value instanceof ChronosDate) {
-            if ($value instanceof NativeDateTime) {
-                $value = clone $value;
-            }
-
-            if (!$value instanceof ChronosDate) {
-                $value = $value->setTime(0, 0, 0);
-            }
-
+        if ($value instanceof $this->_className) {
             return $value;
         }
 
+        /** @phpstan-ignore-next-line */
+        if ($value instanceof DateTimeInterface || $value instanceof ChronosDate) {
```

Comment:
I think since class name always extends ChronosDate, this would never be need to be checked, right?

Reply:
While practically unlikely, it's possible that someone passes a `ChronosDate` to the method when `$className` is set to `Cake\I18n\Date` (i.e. using the full framework). In that case this check would ensure consistent return type by converting `ChronosDate` instance to `Cake\I18n\Date` instance.

Reply:
Ah, I understand. This should work for now.